### PR TITLE
feat(ENG-69): add core accrual types and date math utilities

### PR DESCRIPTION
**Pull request by Connorbelez**
feat(ENG-69): add core accrual types and date math utilities

Foundation for WS6 Accrual & Dispersal Engine. Creates pure computation
layer with Actual/365 day-count convention, full float precision, and
adapted field names (lenderId not investorId). 37 unit tests passing.

Co-Authored-By: Claude Opus 4.6 (1M context) <noreply@anthropic.com>

fix(ENG-69): tighten AccrualResult period type and guard daysBetween

AccrualResult.periods[].toDate is now `string` (not `string | null`)
since results are always clipped to a closed query range. daysBetween
throws on inverted arguments to prevent silent negative accruals in
downstream consumers.

Co-Authored-By: Claude Opus 4.6 (1M context) <noreply@anthropic.com>

## Summary by CodeRabbit

* **New Features**
  * Added interest accrual calculation engine with support for ownership periods, proportional accrual tracking, and date range queries.
  * Added date utility functions for calendar calculations including day counting, date advancement, and date comparisons.

* **Tests**
  * Added comprehensive test suite covering date calculations, accrual formulas, edge cases, and multi-period aggregation.

* **Documentation**
  * Added technical specifications documenting type definitions, mathematical formulas, and implementation details.

### CODE REVIEW

File: convex/accrual/interestMath.ts [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+		total += calculatePeriodAccrual(
+			annualRate,
+			period.fraction,
+			principalBalance,
+			days
+		);
+	}
+
+	return total;
+}
```
[the remark this answers is not in the document]
Comment (Connorbelez) [reply convex/accrual/interestMath.ts]: Fixed in 7ba80c0: Added upfront `fromDate > toDate` guard. Same fix as the Copilot thread — `calculateAccrualForPeriods` now throws on inverted query range.
